### ISSUE TITLE
Are dependency version locks reasonable?

### ISSUE THREAD

**Issue by gulping** [the-simian/es6-plato#63]
Some of deep dependency of this package is the legacy.
(ex: [core-js 1.2.7](https://github.com/zloirock/core-js/releases/tag/v1.2.7))
And some direct dependencies are a little old.
greenkeeper is pointing out it.
https://github.com/the-simian/es6-plato/branches/all

npm doesn't upgrade dependencies that specificated without [range](https://github.com/npm/node-semver#caret-ranges-123-025-004) syntax.
Is this what you intended?

Maybe there is an oversight because I haven't touched the inside of this package.
If there's no ruin, upgraded dependencies are preferred (for me).

**Comment by the-simian** [comment]
Good question. Generally I'm trying to keep the dependencies that matter most up to date. For instance lodash is basically up to date, as is the actual es parser, globby, etc. This is what is used in the code-analysis. The most important dep is `typhonjs-escomplex` that's always as up to date as I can be.

Lets groudn the discussion: At this exact moment, here's every version disparity
```
eslint                  4.19.0  →    5.3.0
 eslint-plugin-react      7.7.0  →   7.10.0 
 fs-extra                 4.0.3  →    7.0.0 
 lodash                  4.17.9  →  4.17.10 
 grunt-casper             0.2.6  →    0.4.2 
 grunt-contrib-nodeunit  ~1.0.0  →   ~2.0.0
 grunt-contrib-uglify    ~3.3.0  →   ~3.4.0 
 grunt-contrib-watch     ~1.0.0  →   ~1.1.0 
```

So its probably good to update eslint (and I will). Lodash is just a tiny tiny bit behind, same with the eslint-plugin-react. 

The dependencies that are really out of date are fs-extra and grunt. I did not initially put these dependencies in the project (these were from the initial fork from plato). I tried updating Grunt once, and it basically broke everything. If I am going to fix a broken grunt, I might as well convert the gruntfiles to something much better like npm and webpack. And tbh, if I'm going to do that, I'd rather rewrite a lot of the generated report as a different kind of app entirely. I think I recall FS-extra breaking when updated as well, but I might look again. 

The point is the parts of the project that really, really need to be up to date - that's the stuff that parses the code, and enforces the linter; stuff like that. I am keeping them up. The dependencies that are *not* up to date are mostly related to spitting out the report itself, which isn't as critical. It won't affect the quality of the report or the scoring or the integration with your project - that's mostly a nice-to-have.

That said, at the very least updating some of these is actionable, so I'm going to do it.

As for your question about the range syntax - I've stopped doing that in most projects entirely. I've been burned too many times by too many maintainers that do not get how semver works. I encourage you to do the same thing and use a tool like `npm-check-updates` to manage dependency updates. I 'd trust lodash out of that list (John is a top notch dev who respects semver convention), but there's others that might be more fragile.  Therefore, I'm locking the dependencies because for this project I'm favoring reproduce-able stability over all else. I'm not budging on this. When I open up version ranges I really consistently get bug reports that are a side effect of child modules, and out of my control. This means its my responsibility to work out all the version ranges that's dependable (pun intended). If you download the project right now, as-is - it *will* work. That's my reason for locking ranges, and its going to stay that way.

I will take this as a call to action to update the dependencies that I can, though. Thanks for your feedback, I hope the tool is working well for you.
